var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -384,6 +398,27 @@ def url_for_version(self, version):
         url = "https://github.com/trilinos/Trilinos/archive/trilinos-release-{0}.tar.gz"
         return url.format(version.dashed)
 
+    def setup_dependent_run_environment(self, env, dependent_spec):
+        if '+cuda' in self.spec:
+            # currently Trilinos doesn't perform the memory fence so
+            # it relies on blocking CUDA kernel launch. This is needed
+            # in case the dependent app also run a CUDA backend via Trilinos
+            env.set('CUDA_LAUNCH_BLOCKING', '1')
+
+    def setup_dependent_package(self, module, dependent_spec):
+        try:
+            self.spec.kokkos_cxx = self.spec["kokkos-nvcc-wrapper"].kokkos_cxx
+        except Exception:
```

Comment:
Don't ever catch `Exception` without rethrowing it: what exception did you see that requires this? Shouldn't you just wrap in `if '+wrapper' in spec`?

Reply:
This is in your official release of kokkos/package.py. I simply copy and paste it. Please make concrete suggestion so that I don't have to guess what the expected behavior is.

Reply:
Sorry @Char-aznable, I didn't realize this was a copy-paste issue. Do you know what happens if you leave it off? Or what exceptions the code raises and under what circumstances?

@jjwilke it looks like you authored the original lines in the kokkos recipe in 598c233f78 ; can you explain what the `kokkos_cxx` and exception here are for?

Reply:
Nope. Not sure why I would have done that instead of checking for the "+wrapper" variant. Probably I was just trying to figure out a way to check for a particular package as a dependency (rather than a variant) and couldn't find a better way.

I actually can't believe this got approved and merged. Usually the Spack team flags awfulness like this and tells me a better way to do it.

Reply:
@jjwilke Do you think this `self.spec.kokkos_cxx` is not necessary at all for dependent package? Do you have a fix?

Reply:
The `kokkos_cxx` is necessary for dependent packages if people rely on `nvcc_wrapper`.  I think just checking for the "+wrapper" variant is good enough and getting rid of the exception.

Reply:
Actually I'm not sure if this block of code is needed here  at all: downstream codes that need to  use the `nvcc_wrapper` can use `self.spec["kokkos-nvcc-wrapper"].kokkos_cxx`; the code here allows them to use `self.spec["trilinos"].kokkos_cxx` which I don't think is useful. It's certainly not needed for the `setup-build_environment` code below.

Reply:
`kokkos-nvcc-wrapper` won't be part of the spec if it's a non-CUDA build.

Reply:
Sure, but the same is true here, right? `spec['trilinos']` isn't going to have a `kokkos_cxx` attribute if it's not building with CUDA.

Reply:
Sorry, the intended behavior is that `spec['kokkos'].kokkos_cxx` is always valid regardless. Entirely possible that it is not implemented correctly.  This is all intended to work exactly the same way the MPI compilers do in Spack.

Reply:
Never mind, I'm being a dum-dum; I forgot the `except`. Changing this to an `if '+wrapper' in spec: ... else: ...` is the way to go here.

Reply:
OK. This is fixed now